dashboard/src/routes/hardware/route.tsx

```diff
@@ -1,14 +1,20 @@
-import { createFileRoute } from '@tanstack/react-router';
+import { createFileRoute, stripSearchParams } from '@tanstack/react-router';
 import { z } from 'zod';
 
 import { makeZIntervalInDays } from '@/types/general';
 import { DEFAULT_HARDWARE_INTERVAL_IN_DAYS } from '@/utils/constants/hardware';
 
+const defaultValues = {
+  intervalInDays: DEFAULT_HARDWARE_INTERVAL_IN_DAYS,
+  hardwareSearch: '',
+};
+
 const zHardwareSchema = z.object({
   intervalInDays: makeZIntervalInDays(DEFAULT_HARDWARE_INTERVAL_IN_DAYS),
-  hardwareSearch: z.string().optional().catch(undefined),
+  hardwareSearch: z.string().optional().catch(''),
```

Comment:
from what I'm understanding with stripSerach poarameters you don't need `.optional` anymore

Reply:
Done